jgiven-junit5/src/main/java/com/tngtech/jgiven/junit5/JGivenExtension.java

```diff
@@ -35,61 +35,62 @@
-        BeforeEachCallback,
-        AfterEachCallback {
+    TestInstancePostProcessor,
+    BeforeAllCallback,
+    AfterAllCallback,
+    BeforeEachCallback,
+    AfterEachCallback {
 
-    private static final Namespace NAMESPACE = Namespace.create( "com.tngtech.jgiven" );
+    private static final Namespace NAMESPACE = Namespace.create("com.tngtech.jgiven");
 
     private static final String REPORT_MODEL = "report-model";
 
     @Override
-    public void beforeAll( ExtensionContext context ) throws Exception {
+    public void beforeAll(ExtensionContext context) {
         ReportModel reportModel = new ReportModel();
-        reportModel.setTestClass( context.getTestClass().get() );
-        if( !context.getDisplayName().equals( context.getTestClass().get().getSimpleName() ) ) {
-            reportModel.setName( context.getDisplayName() );
+        reportModel.setTestClass(context.getTestClass().get());
+        if (!context.getDisplayName().equals(context.getTestClass().get().getSimpleName())) {
+            reportModel.setName(context.getDisplayName());
         }
-        context.getStore( NAMESPACE ).put( REPORT_MODEL, reportModel );
+        context.getStore(NAMESPACE).put(REPORT_MODEL, reportModel);
 
-        AbstractJGivenConfiguration configuration = ConfigurationUtil.getConfiguration( context.getTestClass().get() );
-        if( configuration.getTagConfiguration( Tag.class ) == null ) {
-            configuration.configureTag( Tag.class )
-                .description( "JUnit 5 Tag" )
-                .color( "orange" );
+        AbstractJGivenConfiguration configuration = ConfigurationUtil.getConfiguration(context.getTestClass().get());
+        if (configuration.getTagConfiguration(Tag.class) == null) {
+            configuration.configureTag(Tag.class)
+                .description("JUnit 5 Tag")
+                .color("orange");
         }
     }
 
     @Override
-    public void afterAll( ExtensionContext context ) throws Exception {
+    public void afterAll(ExtensionContext context) {
         ScenarioHolder.get().removeScenarioOfCurrentThread();
-        new CommonReportHelper().finishReport( (ReportModel) context.getStore( NAMESPACE ).get( REPORT_MODEL ) );
+        new CommonReportHelper().finishReport((ReportModel) context.getStore(NAMESPACE).get(REPORT_MODEL));
     }
 
     @Override
-    public void beforeEach( ExtensionContext context ) throws Exception {
-        getScenario().startScenario( context.getTestClass().get(), context.getTestMethod().get(),
-            ArgumentReflectionUtil.getNamedArgs( context ) );
+    public void beforeEach(ExtensionContext context) {
+        getScenario().startScenario(context.getTestClass().get(), context.getTestMethod().get(),
+            ArgumentReflectionUtil.getNamedArgs(context));
     }
 
     @Override
-    public void afterEach( ExtensionContext context ) throws Exception {
+    public void afterEach(ExtensionContext context) throws Exception {
```

Comment:
Is this throwing of an exception then still necessary? Or would it lead to more refactoring?

Reply:
Well we throw an Exception in ln 91. So we need to declare that we do.